iOS/UI/Reader/Page/ReaderPageView.swift

```diff
@@ -136,16 +136,27 @@ class ReaderPageView: UIView {
             )
         }
 
-        let success: Bool
-
-        do {
-            _ = try await ImagePipeline.shared.image(for: request, delegate: self).image
-            success = true
-        } catch {
-            success = false
-        }
+        imageTask = ImagePipeline.shared.loadImage(
+            with: request,
+            progress: { [weak self] _, completed, total in
+                guard let self else { return }
+                self.progressView.setProgress(value: Float(completed) / Float(total), withAnimation: false)
+            },
+            completion: { [weak self] result in
+                guard let self else { return }
+                switch result {
+                case .success(let response):
+                    imageView.image = response.image
+                    fixImageSize()
+                    completion?(true)
+                case .failure:
+                    completion?(false)
+                }
+                progressView.isHidden = true
+            }
+        )
 
-        return success
+        return true
```

Comment:
this will break things since the function is supposed to wait until the image loads; if there's an async version of `loadImage`, you could use that, otherwise use a `withCheckedContinuation` and return with that

Reply:
There's indeed an async function but it returns a new `AsyncImageTask` that doesn't have a `state` variable so I'll use `withCheckedContinuation`. Thanks